app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -151,24 +152,43 @@ function computeRequiredDeposit(certificates: Array<TxCertificate>): Lovelace {
   return certificates.reduce((acc, {type}) => acc + CertificateDeposit[type], 0) as Lovelace
 }
 
-export const computeMinUTxOLovelaceAmount = (
-  address: Address,
-  coins: Lovelace,
-  tokens: Token[]
-): Lovelace => {
+export const computeMinUTxOLovelaceAmount = (tokens: Token[]): Lovelace => {
+  // based on https://github.com/input-output-hk/cardano-ledger-specs/blob/master/doc/explanations/min-utxo.rst
+  const quot = (x: number, y: number) => Math.floor(x / y)
+  const roundupBytesToWords = (x: number) => quot(x + 7, 8)
   // TODO: this to network config or constants
-  const adaOnlyUTxOSize = 74 // TODO: this is too much
   const minUTxOValue = 1000000
-  if (tokens.length === 0) return minUTxOValue as Lovelace
-  const output: TxOutput = {
-    isChange: false,
-    address,
-    coins,
-    tokens,
+  // NOTE: should be 2, but a bug in Haskell set this to 0
+  const coinSize = 0
+  const txOutLenNoVal = 14
+  const txInLen = 7
+  const utxoEntrySizeWithoutVal = 6 + txOutLenNoVal + txInLen // 27
+
+  // NOTE: should be 29 but a bug in Haskell set this to 27
+  const adaOnlyUtxoSize = utxoEntrySizeWithoutVal + coinSize
+
+  const distinctAssets = distinct(tokens.map(({assetName}) => assetName))
+
+  const numAssets = distinctAssets.length
+  const numPIDs = distinct(tokens.map(({policyId}) => policyId)).length
+
+  const sumAssetNameLengths = distinctAssets.reduce(
+    (acc, assetName) => acc + Math.max(Buffer.from(assetName, 'hex').byteLength, 1),
+    0
+  )
+
+  const pidSize = 28
```

Comment:
better be explicit and name it `policyIdSize`, a reader without context may get confused and think first of some "process id" or other totally misleading/unrelated concept

Reply:
Sure, np, but I followed the specification linked above and wanted to be consistent with it as much as possible. A possible reader without context would need to check that specification anyway. Either way, I ll rename it to `policyIdSize`.

Reply:
Anyway, I renamed it to `policyIdSize` and left comment stating that it represents `pid` from the specification.